pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -520,47 +562,75 @@ func (p *PoolManager) vmWaitingCleanupLook() {
 	for _ = range c {
 		p.poolMutex.Lock()
 
-		configMap, err := p.kubeClient.CoreV1().ConfigMaps(p.managerNamespace).Get(context.TODO(), names.WAITING_VMS_CONFIGMAP, metav1.GetOptions{})
-		if err != nil {
-			logger.Error(err, "failed to get config map", "configMapName", names.WAITING_VMS_CONFIGMAP)
-			p.poolMutex.Unlock()
-			continue
-		}
-
-		configMapUpdateNeeded := false
-		if configMap.Data == nil {
-			logger.V(1).Info("the configMap is empty", "configMapName", names.WAITING_VMS_CONFIGMAP, "macPoolMap", p.macPoolMap)
-			p.poolMutex.Unlock()
-			continue
-		}
-
-		for macAddress, allocationTime := range configMap.Data {
-			t, err := time.Parse(time.RFC3339, allocationTime)
+		var configMapUpdateNeeded bool
+		vmtoMacChanges := map[string]macChanges{}
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			configMapUpdateNeeded = false
+			configMap, err := p.kubeClient.CoreV1().ConfigMaps(p.managerNamespace).Get(context.TODO(), names.WAITING_VMS_CONFIGMAP, metav1.GetOptions{})
 			if err != nil {
-				// TODO: remove the mac from the wait map??
-				logger.Error(err, "failed to parse allocation time")
-				continue
+				logger.Error(err, "failed to get config map", "configMapName", names.WAITING_VMS_CONFIGMAP)
+				return nil
+			}
+
+			if configMap.Data == nil {
+				logger.V(1).Info("the configMap is empty", "configMapName", names.WAITING_VMS_CONFIGMAP, "macPoolMap", p.macPoolMap)
+				return nil
 			}
 
 			logger.Info("data:", "configMapName", names.WAITING_VMS_CONFIGMAP, "configMap.Data", configMap.Data, "macPoolMap", p.macPoolMap)
+			for macAddressDashes, cmEntry := range configMap.Data {
```

Comment:
Maybe a dumb question, but why do we keep mac addresses with dashes in the configMap? I mean why not use only colons?

Reply:
I think it will break the configmap yaml format. frankly I never tried it change it.
But it's in my bucket list for refactor changes, after this PR is done.